_posts/2022-01-03-quarkus-aws-cloudwatch_extension.adoc

```diff
@@ -0,0 +1,165 @@
+The Log Handler itself. In the snippet above, in the _LoggingCloudWatchHandlerValueFactory_ we created it already and returned it as a RuntimeValue which we are using in the _LoggingCloudwatchProcessor_ class. That’s the call chain needed to overwrite the existing default log handler.
+
+[source,java]
+----
+class LoggingCloudWatchHandler extends Handler {
+
+    private AWSLogs awsLogs;
+    private String logStreamName;
+    private String logGroupName;
+    private String sequenceToken;
+
+    // ...
+
+    LoggingCloudWatchHandler(AWSLogs awsLogs, String logGroup, String logStreamName, String token) {
+        this.logGroupName = logGroup;
+        this.awsLogs = awsLogs;
+        this.logStreamName = logStreamName;
+        this.sequenceToken = token;
+    }
+
+    @Override
+    public void publish(LogRecord record) {
+
+        // ...
+
+        InputLogEvent logEvent = new InputLogEvent()
+                .withMessage(body)
+                .withTimestamp(System.currentTimeMillis());
+        awsLogs.putLogEvents(request);
+    }
+}
+----
+
+This log handler is a java.util.LogHandler which takes the LogRecord object as a parameter of the publish method which will be called when writing a log in an application. For example like log.info(“I Love Open Source!”);. If configured correctly, this log handler will be called when writing logs. As we want to put the log messages in AWS CloudWatch, we need to add the logic for doing it.
+Therefore we create an InputLogEvent and call _putLogEvents()_ which puts the log message to CloudWatch. That’s basically it.
+
+[NOTE]
+====
+The snippets in this article are a bit shortened, but basically that’s what the extension contains.
+====
+
+Let’s sum it up:
+There is a processor class which initializes the extension, a configuration class which is needed to make the extension configurable, a value factory class which takes these configurations and creates a AWS CloudWatch connection  as well as a custom _LogHandler_ class which pushes each log message to CloudWatch.
+
+After doing all these things, the only thing missing is releasing a version of the extension. This can be done by opening a Pull Request which updates the current-version and _next-version_ entry of the _project.yml_ file in the _.github_ folder. After merging this Pull Request, some GitHub Actions will be triggered which will bring your new version to Maven Central and finally others can use your extension as well :-)
+
+== Summary
+As you can see, creating, implementing and sharing Quarkus extensions with others is actually very easy. So if you have
+an idea of an extension which could be useful for the community, feel free to pitch your idea by creating a new extension proposal issue on the quarkusio/quarkus GitHub Issues section :-)
+
+In case you have questions, suggestions or something else, please feel free to contact me on Twitter.
```

Comment:
I'd suggest to add a link to your Twitter profile

Reply:
I expected to see contact information when clicking on the author name below the title, but this isn't working. Could be something we could add in the future.